src/components/organisms/NewProfileModal/UserProfileModal/UserProfileModalContent/UserProfileModalContent.tsx

```diff
@@ -0,0 +1,244 @@
+
+import { WithId } from "utils/id";
+import { propName, userProfileModalFormProp as formProp } from "utils/propName";
+
+import { useProfileModalFormDefaultValues } from "hooks/useProfileModalFormDefaultValues";
+import { useProfileQuestions } from "hooks/useProfileQuestions";
+import { useShowHide } from "hooks/useShowHide";
+
+import { updateUserProfile } from "pages/Account/helpers";
+
+import { UserProfileModalButtons } from "components/organisms/NewProfileModal/components/buttons/UserProfileModalButtons";
+import { ProfileModalEditBasicInfo } from "components/organisms/NewProfileModal/components/header/ProfileModalEditBasicInfo";
+import { ProfileModalEditLinks } from "components/organisms/NewProfileModal/components/links/ProfileModalEditLinks";
+import { ProfileModalChangePassword } from "components/organisms/NewProfileModal/components/ProfileModalChangePassword";
+import { ProfileModalQuestions } from "components/organisms/NewProfileModal/components/ProfileModalQuestions";
+
+import "../UserProfileModal.scss";
+
+interface Props {
+  user: WithId<User>;
+  venue: WithId<AnyVenue>;
+  onCancelEditing: () => void;
+  isSubmittingState: ReturnType<typeof useShowHide>;
+}
+
+export const arePasswordsNotEmpty = (
+  passwords: UserProfileModalFormDataPasswords
+) => Object.values(pick(passwords, passwordsFields)).some((x) => x);
+
+const passwordsFields: (keyof UserProfileModalFormDataPasswords)[] = [
+  "oldPassword",
+  "newPassword",
+  "confirmNewPassword",
+];
+
+export const UserProfileModalContent: React.FC<Props> = ({
+  venue,
+  user,
+  isSubmittingState,
+  onCancelEditing: turnOffEditMode,
+}) => {
+  const firebase = useFirebase();
+  const firebaseUser = firebase.auth()?.currentUser;
+  const firebaseUserEmail = firebaseUser?.email;
+
+  const { questions, answers } = useProfileQuestions(user, venue.id);
+
+  const {
+    isShown: isSubmitting,
+    show: startSubmitting,
+    hide: finishSubmitting,
+  } = isSubmittingState;
+
+  const defaultValues = useProfileModalFormDefaultValues(
+    user,
+    questions,
+    answers
+  );
+
+  const {
+    register,
+    reset,
+    errors,
+    setError,
+    clearError,
+    watch,
+    handleSubmit,
+    getValues,
+    setValue,
+    control,
+  } = useForm<UserProfileModalFormData>({
+    mode: "onBlur",
+    reValidateMode: "onChange",
+    validateCriteriaMode: "all",
+    defaultValues,
+  });
+
+  const {
+    fields,
+    append: addLink,
+    remove: removeLink,
+  } = useFieldArray<ProfileLink>({
+    control,
+    name: formProp("profileLinks"),
+  });
+
+  const links = fields as WithId<ProfileLink>[];
+
+  const cancelEditing = useCallback(() => {
+    turnOffEditMode();
+    reset();
+  }, [reset, turnOffEditMode]);
+
+  const addLinkHandler = useCallback(() => {
+    addLink({ url: "", title: "" });
+  }, [addLink]);
+
+  const checkOldPassword = useCallback(
+    async (password: string) => {
+      if (!firebaseUserEmail) return;
```

Comment:
```suggestion
      if (!firebaseUserEmail) return;
      
```

Reply:
Extracted separate hook `useCheckOldPassword` and adde blank line there in https://github.com/sparkletown/sparkle/pull/1934/commits/0c31a66aa878f5517a180eb9a170e091bf5e2ae5